OrbitGrpcProtos/capture.proto

```diff
@@ -177,6 +177,50 @@ message AddressInfo {
   }
 }
 
+message GpuQueueSubmission {
+  GpuQueueSubmissionMetaInfo meta_info = 1;
+  repeated GpuSubmitInfo submit_infos = 2;
+  repeated GpuDebugMarker completed_markers = 3;
+  uint32 num_begin_markers = 4;
+}
+
+message GpuQueueSubmissionMetaInfo {
+  int32 thread_id = 1;
+  uint64 pre_submission_cpu_timestamp = 2;
+  uint64 post_submission_cpu_timestamp = 3;
+}
+
+message GpuSubmitInfo {
+  repeated GpuCommandBuffer command_buffers = 1;
+}
+
+message GpuCommandBuffer {
+  uint64 begin_gpu_timestamp_ns = 1;
+  uint64 end_gpu_timestamp_ns = 2;
+}
+
+message GpuDebugMarkerBeginInfo {
+  GpuQueueSubmissionMetaInfo meta_info = 1;
+  uint64 gpu_timestamp_ns = 2;
+}
+
+message GpuDebugMarker {
+  oneof begin_marker_option {
+    GpuDebugMarkerBeginInfo begin_marker = 1;
+  };
```

Comment:
Why `oneof` when you have only one option?

Reply:
That is an purely optional field. Unfortunately, the version of ProtoBuf that we are using, does not support `optional`.

Reply:
That's not the point of `oneof`. In `proto3` all fields are already optional.

Reply:
But then I can't check if that field is there.
https://stackoverflow.com/questions/42622015/how-to-define-an-optional-field-in-protobuf-3/42634681

Reply:
Wait, I was wrong. That was/is the case for primitives. At the beginning, that field should be a primitive. For the others, there are also the ` has` methods.

Reply:
I have to give it to you that it's a bit of a mess for primitives, I forgot about that. But yes, in this case there should be a `has_begin_marker`.

Reply:
Maybe if you want to make it clearer just from the name you could rename `begin_marker` to `optional_begin_marker`, but to me that's not necessary.

Reply:
Changed it accordninly.

Reply:
> Maybe if you want to make it clearer just from the name you could rename begin_marker to optional_begin_marker, but to me that's not necessary.

I think it is fine as it is.